### PR TITLE
Add File System

### PR DESCRIPTION
**Pull request by giulcioffi**
This File System is the one already supported by MKRGSM 1400. Only few functions have been modified/added to use it for MKRNB 1500 and its second stage Bootloader (SBU).

### CODE REVIEW

File: src/Modem.cpp [hunk 1]
```diff
@@ -54,10 +54,11 @@ int ModemClass::begin(bool restart)
     if (!autosense()) {
       return 0;
     }
-
+    /*
```
Comment (aentinger) [inline src/Modem.cpp:57]: Good morning @giulcioffi ☕ I've been wondering if we can't fully delete the code if it's commented out already?

File: src/NBFileUtils.cpp [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+int NBFileUtils::existFile(const String filename)
+{
+    _getFileList();
+    _countFiles();
+
+    String files[_count];
+
+    int num = listFiles(files);
+
+    Serial1.print("Im in existFile. Num of files after listFiles: ");
```
Comment (manchoz) [inline src/NBFileUtils.cpp:62]: Is this a debugging leftover?

File: src/Modem.cpp (hunk 1 above)
Comment (manchoz) [reply src/Modem.cpp]: [FileUtilsJsonFile.ino](https://github.com/arduino-libraries/MKRGSM/blob/master/examples/Tools/FileUtilsJsonFile/FileUtilsJsonFile.ino) as a starting point. (There are other FileUtils examples in the `Tools` directory).

File: src/NBFileUtils.cpp [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+    Serial1.println(num);
```
Comment (manchoz) [inline src/NBFileUtils.cpp:63]: Are these debugging leftovers?

Comment (giulcioffi) [reply src/NBFileUtils.cpp:63]: You're right, the print on Serial1 are debugging leftovers. I'm going to delete them.

**Pull request merged**